Library/Homebrew/command.rb

```diff
@@ -0,0 +1,146 @@
+module Homebrew
+  class Command
+    class << self
+      attr_reader :command_name, :valid_options, :description, :help_output, :man_output, :argv
+    end
+
+    def self.initialize
+      @valid_options = []
+      @description = nil
+      @command_name = nil
+      @help_output = nil
+      @man_output = nil
+      @root_options = []
+      @argv = ARGV
```

Comment:
To build on what Mike said: IMO, it's a good idea to have an `@argv` that's basically just `ARGV.dup`, but it should only be used in the parsing process - a programmer who is writing these DSLs should be interacting with their flags, switches, options, and arguments through methods and objects, like `--foo` -> `opts.foo?` and `--foo=bar` -> `opts[:foo]`.